.github/workflows/quality-check.yaml

```diff
@@ -20,34 +20,56 @@ on:
     types:
       - created
 
-
 jobs:
   check-and-test:
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
-        python-version: [3.7, 3.8]
-        poetry-version: [1.1.6]
+        python-version: [3.10.12]
+        poetry-version: [1.1.12]
     name: Python-${{ matrix.python-version }}
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-python@v2
         with:
           python-version: ${{ matrix.python-version }}
-      # ------------ Install poetry
+
       - name: Setup pip/poetry
         run: |
           pip install -U pip poetry twine
           poetry config virtualenvs.create false
-      # ------------ install tools
+
+      - name: Install Python dependencies
+        run: poetry install
+
+      - name: Configure environment variables for CMake
+        run: |
+          echo "PYTHON_BIN=$(which python)" >> $GITHUB_ENV
```

Comment:
It might be better to define the environment variable in a build step specifically: https://docs.github.com/en/actions/learn-github-actions/variables#defining-environment-variables-for-a-single-workflow

Reply:
not sure what you mean here!

Reply:
Rather than having a separate step to define a new environment variable and piping it into the global environment, the link describes how to add an `env` statement to the step in the github action definition for defining contextual environment variables.